demos/text_detection_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,407 @@
+                           } else {
+                               return std::vector<cv::Point2f>(imgPts);
+                           }
+                        });
+    }
+};
+
+void cropImage(const cv::Mat& image, const std::vector<cv::Point2f>& points,
+               const cv::Size& targetSize, cv::Mat& crop) {
+    const size_t topLeftPointIdx = custom::getTopLeftPointIdx(points);
+    cv::Point2f point0 = points[topLeftPointIdx];
+    cv::Point2f point1 = points[(topLeftPointIdx + 1) % 4];
+    cv::Point2f point2 = points[(topLeftPointIdx + 2) % 4];
+    std::vector<cv::Point2f> from { point0, point1, point2 };
+
+    const float targetWidth  = static_cast<float>(targetSize.width  - 1);
+    const float targetHeight = static_cast<float>(targetSize.height - 1);
+    std::vector<cv::Point2f> to { cv::Point2f(0.0f,        0.0f),
+                                  cv::Point2f(targetWidth, 0.0f),
+                                  cv::Point2f(targetWidth, targetHeight) };
+
+    cv::warpAffine(image, crop, cv::getAffineTransform(from, to), targetSize);
+}
+
+GAPI_OCV_KERNEL(OCVCropLabels, custom::CropLabels) {
+    static void run(const cv::Mat& image,
+                    const std::vector<cv::RotatedRect>& rrs,
+                    const std::vector<size_t>& outDims,
+                    const bool centralCrop,
+                          std::vector<cv::Mat>& out,
+                          std::vector<std::vector<cv::Point2f>>& pts) {
+        cv::Size outSize(outDims[3], outDims[2]);
+        std::vector<int> blobShape(4);
+        std::copy(outDims.begin(), outDims.end(), blobShape.begin());
+        const std::vector<cv::Point2f> imgPts { { 0.0f,              0.0f              },
+                                                { image.cols - 1.0f, 0.0f              },
+                                                { image.cols - 1.0f, image.rows - 1.0f },
+                                                { 0.0f,              image.rows - 1.0f } };
+        out.clear();
+        out.reserve(rrs.size());
+        pts.clear();
+        pts.reserve(rrs.size());
+        cv::Mat crop(outSize,   CV_8UC3,  cv::Scalar(0));
+        cv::Mat blob(blobShape, CV_32FC1, cv::Scalar(0));
+        std::vector<cv::Point2f> points(4);
+        for (auto& rr : rrs) {
+            if (rr.size != cv::Size2f(0, 0)) {
+                rr.points(points.data());
+                cropImage(image, points, outSize, crop);
+            } else {
+                if (centralCrop) {
+                    auto r = centralCropRect(image.cols, image.rows);
+                    points = centralCropPoints(r);
+                    cv::resize(image(r), crop, outSize);
+                } else {
+                    points = imgPts;
+                    cv::resize(image, crop, outSize);
+                }
+            }
+            crop.reshape(1, blobShape).convertTo(blob, CV_32F);
+            out.emplace_back(blob.clone());
```

Comment:
M B create `out = vector<cv::Mat>(rrs.size(), cv::Mat(blobShape, CV_32FC1))`
and use `out[i]` instead blob:
`crop.reshape(1, blobShape).convertTo(out[i], CV_32F);`

Reply:
no, unfortunately, then, without cloning, every `out[i]` would has the shared memory - I mean, all the Mats would be the same
and if handle this, it wouldn't be any shorter

Reply:
out --> labels --> infer. What shared memory do you want handle?

Reply:
so, the problem of this string
`crop.reshape(1, blobShape).convertTo(out[i], CV_32F);`
is that the `crop` is the one which is being assigned to `out[i]`. Without `clone()`. So `out[i]` and `crop`  are bounded

Then, on the next cycle iteration, crop will be changed and assigned to `out[i+1]` - but, `out[i]` will also be changed.

Reply:
Ok. What about reallocation? Have blobs and crops the same data pointers after `convertTo`?

Reply:
`convertTo` does not change the source Mat, so they are different with output after operation.

The problem is/was in `vector`'s constructor which binds all elements

I've figured out a way to do it better (I think) - please take a look on this spot again